backend/api/cms/tests/news/test_queries.py

```diff
@@ -110,6 +121,87 @@ def test_get_news_article(
     }
 
 
+def test_get_news_article_returns_live_revision(
+    graphql_client,
+):
+    user = UserFactory(full_name="marco world")
+    parent = GenericPageFactory()
+    article_1 = NewsArticleFactory(
+        title="Article 1",
+        parent=parent,
+        owner=user,
+        slug="slug",
+        first_published_at=datetime.datetime(2010, 1, 1, 10, 0, 0),
+    )
+    revision_1 = article_1.save_revision()
+    revision_1.publish()
+
+    article_1.title = "Better title"
+    revision_2 = article_1.save_revision(previous_revision=revision_1)
+
+    SiteFactory(hostname="pycon", port=80, root_page=parent)
+
+    query = """query NewsArticle(
+        $hostname: String!,
+        $slug: String!,
+        $language: String!
+    ) {
+        newsArticle(hostname: $hostname, slug: $slug, language: $language) {
+            id
+            title
+            authorFullname
+        }
+    }"""
+
+    response = graphql_client.query(
+        query, variables={"hostname": "pycon", "slug": article_1.slug, "language": "en"}
+    )
+
+    assert response["data"]["newsArticle"]["title"] == "Article 1"
+
+    revision_2.publish()
+
+    response = graphql_client.query(
+        query, variables={"hostname": "pycon", "slug": article_1.slug, "language": "en"}
+    )
+
+    assert response["data"]["newsArticle"]["title"] == "Better title"
+
+
+def test_cannot_get_draft_news_article(
+    graphql_client,
+):
+    user = UserFactory(full_name="marco world")
+    parent = GenericPageFactory()
+    article_1 = NewsArticleFactory(
+        title="Article 1",
+        parent=parent,
+        owner=user,
+        slug="slug",
+        first_published_at=None,
+        live=False,
+    )
+    SiteFactory(hostname="pycon", port=80, root_page=parent)
```

Comment:
Why don't we use pytest-factoryboy? `site_factory` as a fixtre...
Are we using https://github.com/wagtail/wagtail-factories ? 
Probably some factories are not registered, unfortunately, but we can just register on our end...

Reply:
I am migrating away from using factories as fixtures as the test code ends up unnecessarily messy when done. Look at any test code that imports 5 factories, and see it has a long list of parameters.

It gets even messier when you want to maybe add params to the test, and now you need to see where your params would get added.

In general I prefer importing the factory once and using it everywhere it is needed